compiler/damlc/tests/src/DA/Test/Packaging.hs

```diff
@@ -1112,6 +1112,39 @@ dataDependencyTests Tools{damlc,repl,validate,davlDar,oldProjDar} = testGroup "D
               ]
           withCurrentDirectory (tmpDir </> "proj") $ callProcessSilent damlc ["build", "-o", tmpDir </> "proj" </> "proj.dar"]
 
+    , testCaseSteps "Type synonyms over data-dependencies" $ \step -> withTempDir $ \tmpDir -> do
+          step "Building dep"
+          createDirectoryIfMissing True (tmpDir </> "dep")
+          writeFileUTF8 (tmpDir </> "dep" </> "daml.yaml") $ unlines
+              [ "sdk-version: " <> sdkVersion
+              , "name: dep"
+              , "version: 0.1.0"
+              , "source: ."
+              , "dependencies: [daml-prim, daml-stdlib]"
+              ]
+          writeFileUTF8 (tmpDir </> "dep" </> "Foo.daml") $ unlines
+              [ "module Foo where"
+              , "type MyInt = Int"
```

Comment:
Does it make sense to put an apostrophe into the name of the synonym to make sure we're getting the mangling/unmangling right?

Reply:
👍  Done!